libcontainer/cgroups/fs/freezer.go

```diff
@@ -79,6 +79,13 @@ func (s *FreezerGroup) Set(path string, r *configs.Resources) (Err error) {
 				// system.
 				time.Sleep(10 * time.Microsecond)
 			}
+
+			if i%500 == 499 {
+				// should sleep a longer time for
+				// some really very slow machine.
+				time.Sleep(5 * time.Second)
```

Comment:
This is a really very long sleep interval.

Reply:
Yes, for some machines, it may cause frozen failed if not sleep a longer time.

Reply:
> This is a really very long sleep interval.

Yes, for some machines, it may cause frozen failed if not sleep a longer time.

Reply:
In my opinion, I think this like a `cat-mouse` game, when the mouse runs more quickly, the cat needs more stength.
So, in other slow machine, maybe we need 6 seconds or more, I think we have sleep for many times, and have returned an error to the caller when failure. I think the caller should have a retry mechanism when got an error in a slow machine. It's the job of the caller, not in this function, WDYT?

Reply:
> In my opinion, I think this looks like a `cat-is-catching-mouse` game, when the mouse runs more quickly, the cat needs more stength. So, in other slow machine, maybe we need 6 seconds or more, I think we have sleep for many times, and have returned an error to the caller when failure. I think the caller should have a retry mechanism when got an error in a slow machine. It's the job of the caller, not in this function, WDYT?

Indeed, as you mentioned, it's hard to determine the exact sleep time. In my case, when I was running an LLM training container, I needed to checkpoint the container and export a snapshot. A relatively long sleep time was required to read correct value of `freezer.state`, ensuring that the checkpoint was successful. Of course, if necessary, I think we could add a retry mechanism to ensure the program is foolproof.